org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/ui/contentAssist/ContentAssistFragment2.xtend

```diff
@@ -159,11 +160,25 @@ class ContentAssistFragment2 extends AbstractInheritingFragment {
 		// take the non-abstract class signature for the src-gen class in case of !generateStub
 		//  as proposalProviders of sub languages refer to 'grammar.proposalProviderClass',
 		//  see 'getGenProposalProviderSuperClass(...)'
+		generateGenJavaProposalProvider(assignments, remainingRules)
+	}
+
+	/**
+	 * @since 2.23
+	 */
+	protected def GeneratedJavaFileAccess generateGenJavaProposalProvider(List<Assignment> assignments, List<AbstractRule> remainingRules) {
 		val genClass =
 			if (isGenerateStub) grammar.genProposalProviderClass else grammar.proposalProviderClass;
 		
+		val superClass = grammar.genProposalProviderSuperClass
+		generateGenJavaProposalProvider(assignments, remainingRules, genClass, superClass)
+	}
+
+	/**
+	 * @since 2.23
+	 */
+	protected def GeneratedJavaFileAccess generateGenJavaProposalProvider(List<Assignment> assignments, List<AbstractRule> remainingRules, TypeReference genClass, TypeReference superClass) {
```

Comment:
I'd like to have a better name for `remainingRules`, maybe just `rules` or `rulesToComplete` ? Same for the method above.